modules/services/shikane.nix

```diff
@@ -0,0 +1,79 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.shikane;
+  tomlFormat = pkgs.formats.toml { };
+in {
+  meta.maintainers = [ hm.maintainers.therealr5 ];
+  options.services.shikane = {
+    enable = mkEnableOption
+      "shikane, A dynamic output configuration tool that automatically detects and configures connected outputs based on a set of profiles.";
+    package = mkOption {
+      type = types.package;
+      default = pkgs.shikane;
+      defaultText = literalExpression "pkgs.shikane";
+      description = "The package to use for shikane.";
+    };
+    settings = mkOption {
+      type = tomlFormat.type;
+      default = { };
+      example = literalExpression ''
+        {
+          profile = [
+            {
+              name = "external-monitor-default";
+              output = [
+                {
+                  match = "eDP-1";
+                  enable = true;
+                }
+                {
+                  match = "HDMI-A-1";
+                  enable = true;
+                  position = {
+                    x = 1920;
+                    y = 0;
+                  };
+                }
+              ];
+            }
+            {
+              name = "builtin-monitor-only";
+              output = [
+                {
+                  match = "eDP-1";
+                  enable = true;
+                }
+              ];
+            }
+          ];
+        }
+      '';
+      description = ''
+        Configuration written to
+        <filename>$XDG_CONFIG_HOME/shikane/config.toml</filename>.
+        </para><para>
+        See <link xlink:href="https://gitlab.com/w0lff/shikane/-/blob/master/docs/shikane.5.man.md" />
+        for more information.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    xdg.configFile."shikane/config.toml".source =
+      tomlFormat.generate "shikane-config" cfg.settings;
```

Comment:
Perhaps could remove this and instead change the service to contain

```
Service = {
  ExecStart = escapeShellArgs [
    (getExe cfg.package) "--config" (tomlFormat.generate "shikane-config" cfg.settings)
  ];
}
```

That should also fix so that the service is automatically restarted on configuration change.

Reply:
Thought about that too but wasn't sure whether it was good idea regarding the tests of the config files. (In other words I didn't know how I should implement tests for files living only in the nix store)